pkg/controller/expcertificates/readiness/readiness_controller.go

```diff
@@ -148,12 +148,33 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 	crt = crt.DeepCopy()
 	apiutil.SetCertificateCondition(crt, condition.Type, condition.Status, condition.Reason, condition.Message)
 
-	if input.Secret != nil && input.Secret.Data != nil {
+	// set Certificate status fields based on the contents of the x509 certificate
+	// if the certificate is not issued, or the data cannot be decoded, the status
+	// field values are cleared.
+	clearCertificateFields := func() {
```

Comment:
This feels weird to see this in Go... might `sunc clearCertificateFields(*cert) ` not be a better option than to assign a function to a variable on every run?

Reply:
It seemed like it made it easier to follow the code, as this function is only really used to save having to write out those 3 lines twice in this one switch statement.

Otherwise I'll probably just expand out the function myself here, to save the reader having to jump to another part of the file just to see a very uneventful function 😄

Reply:
I would prefer to have it written out, it's just 3 lines :)

Reply:
Done 😄